### PR TITLE
Check order as delivered

### PR DESCRIPTION
**Pull request by Dharmraj-48**
Added Condition to check if order is already delivered then prompt the error message

Screenshot from 2020-03-17 14-48-28

### CODE REVIEW

**Comment by gavindsouza** [comment]
@Dharmraj-48 how many commits are supposed to be a part of this PR?

**Comment by Dharmraj-48** [comment]
commits are fixed for
https://github.com/frappe/erpnext/pull/20966

File: erpnext/selling/doctype/sales_order/sales_order_list.js [hunk 1]
```diff
@@ -47,6 +47,46 @@ frappe.listview_settings['Sales Order'] = {
...
+	},
+
+	onload: function(doclist) {
+		const action = () => {
+			const selected_docs = doclist.get_checked_items();
+			const docnames = doclist.get_checked_items(true);
+
+			if (selected_docs.length > 0) {
+				for (let doc of selected_docs) {
+					debugger;
```
Comment (gavindsouza) [inline erpnext/selling/doctype/sales_order/sales_order_list.js:59]: cleanup required?

```suggestion
```

File: erpnext/selling/doctype/sales_order/sales_order_list.js [hunk 2]
```diff
@@ -47,6 +47,46 @@ frappe.listview_settings['Sales Order'] = {
...
+					console.log(doc.docstatus);
```
Comment (gavindsouza) [inline erpnext/selling/doctype/sales_order/sales_order_list.js:60]:
```suggestion
```

**Review by gavindsouza — CHANGES_REQUESTED** [review]

**Comment by marination** [comment]
1) I think a better approach to avoiding mapping delivered sales orders would be hiding the **Pick List** button under **Create** in a Sales Order if the order is 100% delivered.
2) When it comes to fetching Sales Orders into a Pick List, it can be done from a new **Pick List** itself. A filtered list of Sales Orders is fetched to select so you don't have to check for delivery status again.
3) Adding only a creation provision for Pick List in the List View makes the other actions seem less important. Invoice, Delivery Note, Work Order creation from Sales Order are very frequent uses too. So i wouldn't handle this through List View.

**Pull request closed without merging**